mkosi/__init__.py

```diff
@@ -3676,15 +3678,14 @@ def insert_partition(
 
     MkosiPrinter.print_step("Writing partition...")
 
-    with contextlib.ExitStack() as stack:
-        if args.root_partno == partno:
-            luks_format_root(args, loopdev, False, False, True)
-            dev = stack.enter_context(luks_setup_root(args, loopdev, False, True))
-        else:
-            dev = None
+    if args.root_partno == partno:
+        luks_format_root(args, loopdev, False, False, True)
+        cm = luks_setup_root(args, loopdev, False, True)
+    else:
+        cm = contextlib.nullcontext(partition(loopdev, partno))
 
-        path = dev if dev is not None else partition(loopdev, partno)
-        run(["dd", f"if={blob.name}", f"of={path}", "conv=nocreat,sparse"])
+    with cm as dev:
+        run(["dd", f"if={blob.name}", f"of={dev}", "conv=nocreat,sparse"])
```

Comment:
This doesn't handle the case where `dev` is `None`

Reply:
Ooops, indeed. Should be OK now.